src/Tooltip/Tooltip.stories.tsx

```diff
@@ -36,57 +43,25 @@ storiesOf('molecules|Tooltip', module)
         <Tooltip
           placement={TooltipPlacement.TOP}
           onOutsideClick={({ close }) => close()}
-          renderTrigger={({ ref }: { ref: Ref<HTMLButtonElement> }) => (
-            <Icon ref={ref}>
-                <MdInfo />
-            </Icon>
-          )}>
-          <div style={{ width: 300 }}>
-            {Faker.lorem.paragraphs(2)}
-          </div>
+          TriggerComponent={TriggerIcon}
+        >
+          <div style={{ width: 300 }}>{Faker.lorem.paragraphs(2)}</div>
         </Tooltip>
       </div>
     </Flex>
   ))
   .add('controlled', ({ state, storeState }: any) => (
     <Flex justifyContent='center' alignItems='center' bg='lightgray' height={500}>
       <div>
-        <Tooltip
-          open={state.one}
-          placement={TooltipPlacement.TOP}
-          renderTrigger={({ ref }: { ref: Ref<HTMLButtonElement> }) => (
-              <Button
-                ref={ref}
-                onClick={() => storeState({ one: !state.one })}
-              >
-                Click on me
-              </Button>
-          )}>
-          <div style={{ width: 300 }}>
-            {Faker.lorem.paragraphs(2)}
-          </div>
-        </Tooltip>
-      </div>
-    </Flex>
-  ))
-  .add('controlled. close on outside click', ({ state, storeState }: any) => (
-    <Flex justifyContent='center' alignItems='center' bg='lightgray' height={500}>
-      <div>
-        <Tooltip
+        <Tooltip<ControlledTriggerProps>
```

Comment:
I must say I'm not a huge fan of passing generics this way. I'm very confused by the TS generics syntax in XML. Is there another way?

Reply:
It's not mandatory. The compiler will try to infer the type